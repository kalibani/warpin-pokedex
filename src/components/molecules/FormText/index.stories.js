// Stories FormText Component
// --------------------------------------------------------
import React, { Fragment } from 'react';
import { storiesOf } from '@storybook/react';
import FormText from './index';
import Notes from './readme.md';

storiesOf('Molecules', module).add('FormText', () => (
  <FormText />
), {
  info: { inline: true, header: false, text: Notes }
});
