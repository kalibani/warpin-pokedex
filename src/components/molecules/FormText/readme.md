## FormText Component
FormText Component is ....