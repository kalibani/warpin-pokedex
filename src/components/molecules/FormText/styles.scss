/* ==========================================================================
Styles for FormText component
========================================================================== */

// import mixin and variables
@import "../../../assets/styles/variables";

// Start styles FormText component
.m-form-text {
  width: 100%;
  position: relative;
  left: 27px;
  .form-text-wrapper {
    form {
      display: flex;
      .form-control {
        border-radius: 30px;
        width: 100%;
        height: 46px;
        line-height: 46px;
        font-size: 1.1em;
        color: rgba(0,0,0,0.5);
        border: none;
        padding: 10px 20px;
        position: relative;
        // left: 50px;
        &::placeholder {
          color: $silver;
          font-size: 16px;
        }
      }
      .button-form-text {
        height: 46px;
        padding: 10px 26px;
        border: none;
        border-radius: 30px;
        position: relative;
        right: 50px;
        background-color: #032241;
        color: $white;
        &:hover {
          color: $silver;
        }
        svg {
          position: relative;
          left: 10px;
          top: 2px;
        }
      }
    }
  }

}
