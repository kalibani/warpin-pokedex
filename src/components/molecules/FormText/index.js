/* eslint-disable import/no-cycle */
// FormText Component
// --------------------------------------------------------

import React, { useCallback } from 'react';
import { shallowEqual, useSelector, useDispatch } from 'react-redux';
import classname from 'classnames';
import { Input, Button } from 'components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faSearch
} from '@fortawesome/free-solid-svg-icons';
import './styles.scss';
import { searchPokemonByTypes, setKeyword } from 'stores/actions/pokemon';


const FormText = () => {
  const { keyword } = useSelector(({ pokemon }) => ({
    keyword: pokemon.keyword
  }), shallowEqual);

  const dispatch = useDispatch();

  const searchByTypes = useCallback((e) => {
    e.preventDefault();
    dispatch(searchPokemonByTypes());
  }, [dispatch]);

  const handleChange = useCallback(({ target: { value } }) => dispatch(setKeyword(value)),
    [dispatch]);
  const classNames = classname('m-form-text');
  return (
    <div className={classNames}>
      <div className="form-text-wrapper">
        <form id="form" onSubmit={searchByTypes}>
          <Input
            id="keyword"
            name="keyword"
            type="text"
            placeholder="Search for a types of Pokemon......"
            value={keyword}
            className="form-control"
            onChange={handleChange}
          />
          <Button
            type="submit"
            className="button-form-text"
          >
            <span>
              Search
            </span>
            <FontAwesomeIcon icon={faSearch} />
          </Button>
        </form>
      </div>
    </div>
  );
};

export default FormText;
