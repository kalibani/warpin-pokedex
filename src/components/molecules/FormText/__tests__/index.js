// Test FormText Component
// --------------------------------------------------------

import React from 'react';
import FormText from '../../index';
import renderer from 'react-test-renderer';

test('Input is match the snapshot', () => {
const component = renderer.create(
FormText,
);
let tree = component.toJSON();
expect(tree).toMatchSnapshot();
});
