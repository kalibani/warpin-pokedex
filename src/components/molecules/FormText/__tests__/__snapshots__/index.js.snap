// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input is match the snapshot 1`] = `null`;
